nmdc_ms_metadata_gen/metadata_generator.py

```diff
@@ -545,844 +545,869 @@ def generate_chromatography_configuration(
+        when the processed files are iterated over in the run method.
 
         """
-        nmdc_id = self.id_pool.get_id(
-            nmdc_type=NmdcTypes.MassSpectrometry,
-            client_id=CLIENT_ID,
-            client_secret=CLIENT_SECRET,
-        )
+        if incremeneted_id is None:
+            # If no incremented id is provided, mint a new one
+
+            nmdc_id = (
+                self.id_pool.get_id(
+                    nmdc_type=NmdcTypes.MetabolomicsAnalysis,
+                    client_id=CLIENT_ID,
+                    client_secret=CLIENT_SECRET,
+                )
+                + ".1"
+            )
 
         data_dict = {
-            "id": nmdc_id,
-            "name": file_path.stem,
-            "description": self.mass_spec_desc,
-            "add_date": datetime.now().strftime("%Y-%m-%d"),
-            "eluent_introduction_category": self.mass_spec_eluent_intro,
-            "has_mass_spectrometry_configuration": mass_spec_configuration_id,
-            "has_chromatography_configuration": lc_config_id,
-            "analyte_category": self.analyte_category,
-            "instrument_used": instrument_id,
-            "has_input": [sample_id],
-            "has_output": [raw_data_id],
-            "associated_studies": study_id,
+            "id": incremeneted_id if incremeneted_id is not None else nmdc_id,
+            "name": f"{self.workflow_analysis_name} for {raw_data_name}",
+            "description": self.workflow_description,
             "processing_institution": processing_institution,
-            "start_date": start_date,
-            "end_date": end_date,
-            "instrument_instance_specifier": instrument_instance_specifier,
-            "type": NmdcTypes.MassSpectrometry,
+            "execution_resource": cluster_name,
+            "git_url": self.workflow_git_url,
+            "version": self.workflow_version,
+            "was_informed_by": data_gen_id_list,
+            "has_input": [raw_data_id, parameter_data_id],
+            "has_output": [processed_data_id],
+            "started_at_time": "placeholder",
+            "ended_at_time": "placeholder",
+            "type": type,
+            "metabolomics_analysis_category": self.workflow_category,
         }
 
         if calibration_id is not None:
-            data_dict["generates_calibration"] = calibration_id
+            data_dict["uses_calibration"] = calibration_id
+
+        if metabolite_identifications is not None:
+            data_dict["has_metabolite_identifications"] = metabolite_identifications
         data_dict = self.clean_dict(data_dict)
-        mass_spectrometry = nmdc.DataGeneration(**data_dict)
+        metab_analysis = nmdc.MetabolomicsAnalysis(**data_dict)
 
-        return mass_spectrometry
+        return metab_analysis
 
-    def generate_processed_sample(
-        self, data: dict, CLIENT_ID: str, CLIENT_SECRET: str
-    ) -> nmdc.ProcessedSample:
+    def update_outputs(
```

Comment:
This method doesn't make sense on the NMDCMetadataGenerator and should be moved down to NMDCWorkflowMetadataGenerator, but otherwise, these look great.  Thanks for taking the time to move all those methods.

Reply:
(the update_outputs) one, in case it's not clear.